queries/c/textobjects.scm

```diff
@@ -93,25 +153,38 @@
   (_) @statement.outer)
 
 ((parameter_list
-  "," @_start . (parameter_declaration) @parameter.inner)
- (#make-range! "parameter.outer" @_start @parameter.inner))
+  "," @_start
+  .
+  (parameter_declaration) @parameter.inner)
+  (#make-range! "parameter.outer" @_start @parameter.inner))
+
 ((parameter_list
-  . (parameter_declaration) @parameter.inner . ","? @_end)
- (#make-range! "parameter.outer" @parameter.inner @_end))
+  .
+  (parameter_declaration) @parameter.inner
+  .
+  ","? @_end)
+  (#make-range! "parameter.outer" @parameter.inner @_end))
 
 ((argument_list
-  "," @_start . (_) @parameter.inner)
- (#make-range! "parameter.outer" @_start @parameter.inner))
+  "," @_start
+  .
+  (_) @parameter.inner)
+  (#make-range! "parameter.outer" @_start @parameter.inner))
+
 ((argument_list
-  . (_) @parameter.inner . ","? @_end)
- (#make-range! "parameter.outer" @parameter.inner @_end))
+  .
+  (_) @parameter.inner
+  .
+  ","? @_end)
+  (#make-range! "parameter.outer" @parameter.inner @_end))
 
 (number_literal) @number.inner
 
 (declaration
-  declarator: (init_declarator
-    declarator: (_) @assignment.lhs
-    value: (_) @assignment.rhs) @assignment.inner) @assignment.outer
+  declarator:
+    (init_declarator
+      declarator: (_) @assignment.lhs
+      value: (_) @assignment.rhs) @assignment.inner) @assignment.outer
```

Comment:
This should not have been the case . I will look at it later today. Seems like some edge case got missed out on